client/app/src/main/java/com/example/criengine/Activities/NonOwnerBookViewActivity.java

```diff
@@ -90,19 +107,27 @@ protected void onCreate(Bundle savedInstanceState) {
      * Check if the book can be requested. Sets the button to the appropriate text.
      */
     public void checkIfAvailable() {
-        // TODO: Need to retrieve user profile & check if it already exists in list of requesters.
-        if (!book.getStatus().equals("borrowed") && !book.getStatus().equals("accepted")) {
+        if (book.getRequesters().contains(userProfile.getUserID())) {
+            // The user has already requested this book.
+            requestBookButton.setEnabled(false);
+            requestBookButton.setText("Request Sent");
+        } else if (!book.getStatus().equals("borrowed") && !book.getStatus().equals("accepted")) {
             requestBookButton.setEnabled(true);
             requestBookButton.setText("Request This Book");
             requestBookButton.setOnClickListener(new View.OnClickListener() {
```

Comment:
If this is a general request book button, then it shouldn't be set within this function, but when it's created (onCreate) unless we're changing the listener in other places